src/main/commandServer/httpCommandServer.ts

```diff
@@ -176,6 +181,88 @@ export class HttpCommandServer {
     return undefined;
   }
 
+  protected diagnosticCategories(request: http.IncomingMessage, response: http.ServerResponse, context: commandContext): Promise<void> {
+    const settings = this.commandWorker.getDiagnosticCategories(context);
+
+    if (settings) {
+      console.debug('diagnosticCategories: succeeded 200');
+      response.writeHead(200, { 'Content-Type': 'text/plain' });
+      response.write(settings);
+    } else {
+      console.debug('diagnosticCategories: failed 404');
+      response.writeHead(404, { 'Content-Type': 'text/plain' });
+      response.write('No diagnostic categories found');
+    }
+
+    return Promise.resolve();
+  }
+
+  protected diagnosticIDsForCategory(request: http.IncomingMessage, response: http.ServerResponse, context: commandContext): Promise<void> {
+    const url = new URL(`http://${ request.url }`);
+    const searchParams = url.searchParams;
+    const category = searchParams.get('category');
+
+    if (!category) {
+      console.debug('diagnostic_ids: failed 400');
+      response.writeHead(400, { 'Content-Type': 'text/plain' });
+      response.write('diagnostic_ids: no  category specified');
+
+      return Promise.resolve();
+    }
+    const categories = this.commandWorker.getDiagnosticIdsByCategory(category, context);
+
+    if (categories) {
+      console.debug('diagnostic_ids: succeeded 200');
+      response.writeHead(200, { 'Content-Type': 'text/plain' });
```

Comment:
This should be `application/json` of some sort.

Reply:
Again, I haven't seen a need for this yet